app/ide-desktop/lib/performance-logger/src/performance_logger.ts

```diff
@@ -0,0 +1,245 @@
+export enum LogLevel {
+    Section = 'SECTION',
+    Task = 'TASK',
+    Detail = 'DETAIL',
+    Debug = 'DEBUG',
+}
+
+/// Delimiter used to to encode information in the `PerformanceEntry` name.
+const MESSAGE_DELIMITER = '//'
+
+const ENABLE_SECTION_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('section')
+const ENABLE_TASK_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('task')
+const ENABLE_DETAIL_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('detail')
+const ENABLE_DEBUG_PROFILING = process.env.PROFILING_LEVEL.toLowerCase().includes('debug')
+
+/// Check whether the given log level should perform any logging activity.
+function log_level_is_active(log_level: LogLevel): boolean {
+    switch (log_level) {
+        case LogLevel.Section:
+            return ENABLE_SECTION_PROFILING
+        case LogLevel.Task:
+            return ENABLE_TASK_PROFILING
+        case LogLevel.Detail:
+            return ENABLE_DETAIL_PROFILING
+        case LogLevel.Debug:
+            return ENABLE_DEBUG_PROFILING
+    }
+}
+
+/// Return a string that encodes the given log level and name for a mark that indicates the start of
+/// an interval.
+function start_interval_label(log_level: LogLevel, interval_name: string): string {
+    return log_level.toString() + MESSAGE_DELIMITER + interval_name + MESSAGE_DELIMITER + 'start'
```

Comment:
Use string interpolations instead. T?hey are easier to read and faster to compute

Reply:
Replaced this with `MESSAGE_DELIMITER .join` which seems even more fitting, and other string concatenations with string interpolations.